go/oasis-node/cmd/common/metrics/metrics.go

```diff
@@ -6,32 +6,64 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	"os"
+	"path/filepath"
 	"strings"
+	"sync"
 	"time"
 
+	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/procfs"
 	flag "github.com/spf13/pflag"
 	"github.com/spf13/viper"
 
 	"github.com/oasislabs/oasis-core/go/common/service"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common"
 )
 
 const (
-	cfgMetricsMode              = "metrics.mode"
-	cfgMetricsAddr              = "metrics.address"
-	cfgMetricsPushJobName       = "metrics.push.job_name"
-	cfgMetricsPushInstanceLabel = "metrics.push.instance_label"
-	cfgMetricsPushInterval      = "metrics.push.interval"
-
-	metricsModeNone = "none"
-	metricsModePull = "pull"
-	metricsModePush = "push"
+	CfgMetricsMode         = "metrics.mode"
+	CfgMetricsAddr         = "metrics.address"
+	CfgMetricsPushJobName  = "metrics.push.job_name"
+	CfgMetricsPushLabels   = "metrics.push.labels"
+	CfgMetricsPushInterval = "metrics.push.interval"
+
+	MetricsModeNone = "none"
+	MetricsModePull = "pull"
+	MetricsModePush = "push"
+
+	// getconf CLK_TCK
+	ClockTicks = 100
 )
 
-// Flags has the flags used by the metrics service.
-var Flags = flag.NewFlagSet("", flag.ContinueOnError)
+var (
+	// Flags has the flags used by the metrics service.
+	Flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	diskServiceOnce sync.Once
+	memServiceOnce  sync.Once
+	cpuServiceOnce  sync.Once
+	netServiceOnce  sync.Once
+)
+
+// ParseMetricPushLabes is a drop-in replacement for viper.GetStringMapString due to https://github.com/spf13/viper/issues/608.
```

Comment:
The linked issue mentions this may be fixed in 1.6.2? Also see this comment: https://github.com/spf13/viper/issues/608#issuecomment-605762584.

Reply:
It's not. I posted a reply: https://github.com/spf13/viper/issues/608#issuecomment-606014442

Reply:
Maybe rename the method to `GetStringMapString` in that case, since it's a general method and not specific to parsing metric labels.

Also If it could be useful elsewhere the method could be moved into common flags code.

Reply:
Yeah I agree with @ptrus, it seems to be a general method, so consider renaming and moving it to the common flags package.

Reply:
done